zephyr-env.sh: Use $XDG_CONFIG_HOME/zephry/zephyrrc if present

schuam opened the pull request:
According to the [XDG Base Directory Specification] there is a single base directory in which user configuration files should be stored. This helps users to keep their home directory uncluttered. This base directory can be set in the environment variable `$XDG_CONFIG_HOME` and defaults to ${HOME}/.config if $XDG_CONFIG_HOME is not set.

This commit adjusts the zephyr-env.sh script to look for a user configuration file in the following locations:

- $XDG_CONFIG_HOME/zephyr/zephyrrc
- $HOME/zephyrrc
- $HOME/.zephyrrc

The first one that is found will be used.

[XDG Base Directory Specification]: (https://specifications.freedesktop.org/basedir-spec/latest/)

schuam committed:
zephyr-env.sh: Use $XDG_CONFIG_HOME/zephry/zephyrrc if present

According to the [XDG Base Directory Specification] there is a single base
directory in which user configuration files should be stored. This helps
users to keep their home directory uncluttered. This base directory can be
set in the environment variable $XDG_CONFIG_HOME and defaults to
${HOME}/.config if $XDG_CONFIG_HOME is not set.

This commit adjusts the zephyr-env.sh script to look for a user
configuration file in the following locations:

- $XDG_CONFIG_HOME/zephyr/zephyrrc
- $HOME/.config/zephyr/zephyrrc
- $HOME/.zephyrrc

The first one that is found will be used.

[XDG Base Directory Specification]: (https://specifications.freedesktop.org/basedir-spec/latest/)

Signed-off-by: Andreas Schuster <andreas.schuster@schuam.de>

pdgendt commented:
Hi @schuam, I think this could be a welcome addition, could update the documentation to reflect this change? I'd suggest in https://github.com/zephyrproject-rtos/zephyr/blob/eff39fb70ad0bba02c81f6ab6df5bcecde53bbe0/doc/develop/env_vars.rst

schuam committed:
doc: develop: Add XDG_CONFIG_HOME information for zephyrrc

The `zephyr-env.sh` script was adjusted to look for the zephyrrc file in
three locations following the [XDG Base Directory Specification]. The
documentation of the environment variables is now adjusted to reflect that
change.

[XDG Base Directory Specification]: (https://specifications.freedesktop.org/basedir-spec/latest/)

Signed-off-by: Andreas Schuster <andreas.schuster@schuam.de>

schuam commented:
Thanks for your input @pdgendt. I updated the file you mentioned. Please have a look. Any feedback is welcome.
